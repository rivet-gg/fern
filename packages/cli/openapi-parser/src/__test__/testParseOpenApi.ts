import { AbsoluteFilePath, join, RelativeFilePath } from "@fern-api/fs-utils";
import { CONSOLE_LOGGER } from "@fern-api/logger";
import { serialization } from "@fern-api/openapi-ir-sdk";
import { createMockTaskContext } from "@fern-api/task-context";
import { parse, SpecImportSettings } from "../parse";

const FIXTURES_PATH = join(AbsoluteFilePath.of(__dirname), RelativeFilePath.of("fixtures"));

// eslint-disable-next-line jest/no-export
export function testParseOpenAPI(
    fixtureName: string,
    openApiFilename: string,
    asyncApiFilename?: string,
    settings?: SpecImportSettings
): void {
    // eslint-disable-next-line jest/valid-title
    describe(fixtureName, () => {
        it("parse open api", async () => {
            const absolutePathToOpenAPI = join(
                FIXTURES_PATH,
                RelativeFilePath.of(fixtureName),
                RelativeFilePath.of(openApiFilename)
            );
            const absolutePathToAsyncAPI =
                asyncApiFilename != null
                    ? join(FIXTURES_PATH, RelativeFilePath.of(fixtureName), RelativeFilePath.of(asyncApiFilename))
                    : undefined;
            const specs = [];
            if (absolutePathToOpenAPI != null) {
                specs.push({
                    absoluteFilepath: absolutePathToOpenAPI,
                    absoluteFilepathToOverrides: undefined,
                    settings
                });
            }
            if (absolutePathToAsyncAPI != null) {
                specs.push({
                    absoluteFilepath: absolutePathToAsyncAPI,
                    absoluteFilepathToOverrides: undefined,
                    settings
                });
            }

            const openApiIr = await parse({
                specs,
                taskContext: createMockTaskContext({ logger: CONSOLE_LOGGER })
            });
            const openApiIrJson = await serialization.OpenApiIntermediateRepresentation.jsonOrThrow(openApiIr, {
                skipValidation: true
            });
            expect(openApiIrJson).toMatchSnapshot();
        });
    });
}
