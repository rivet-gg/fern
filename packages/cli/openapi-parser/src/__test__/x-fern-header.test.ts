import { testParseOpenAPI } from "./testParseOpenApi";

describe("x-fern-header", () => {
    testParseOpenAPI("x-fern-header", "openapi.yml");
});
