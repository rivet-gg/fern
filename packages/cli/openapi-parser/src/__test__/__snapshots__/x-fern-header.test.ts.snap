// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`x-fern-header x-fern-header parse open api 1`] = `
{
  "apiVersion": null,
  "basePath": null,
  "channel": [],
  "description": null,
  "endpoints": [
    {
      "audiences": [],
      "authed": false,
      "availability": null,
      "description": null,
      "errors": {},
      "examples": [],
      "generatedRequestName": "UserGetStreamRequest",
      "headers": [],
      "idempotent": null,
      "internal": null,
      "method": "GET",
      "operationId": "get_stream",
      "pagination": null,
      "path": "/user/{userId}",
      "pathParameters": [
        {
          "availability": null,
          "description": null,
          "name": "userId",
          "schema": {
            "availability": null,
            "description": null,
            "generatedName": "UserGetStreamRequestUserId",
            "groupName": null,
            "nameOverride": null,
            "schema": {
              "default": null,
              "format": null,
              "maxLength": null,
              "minLength": null,
              "pattern": null,
              "type": "string",
            },
            "type": "primitive",
          },
          "variableReference": null,
        },
      ],
      "queryParameters": [],
      "request": {
        "additionalProperties": false,
        "contentType": null,
        "description": null,
        "schema": {
          "additionalProperties": false,
          "allOf": [],
          "allOfPropertyConflicts": [],
          "availability": null,
          "description": null,
          "generatedName": "UserGetStreamRequest",
          "groupName": null,
          "nameOverride": null,
          "properties": [
            {
              "audiences": [
                "public",
              ],
              "availability": null,
              "conflict": {},
              "generatedName": "userGetStreamRequestStream",
              "key": "stream",
              "nameOverride": null,
              "schema": {
                "availability": null,
                "description": null,
                "generatedName": "UserGetStreamRequestStream",
                "groupName": null,
                "nameOverride": null,
                "type": "literal",
                "value": {
                  "type": "boolean",
                  "value": true,
                },
              },
            },
          ],
          "type": "object",
        },
        "type": "json",
      },
      "requestNameOverride": null,
      "response": {
        "description": "",
        "responseProperty": null,
        "schema": {
          "availability": null,
          "description": null,
          "generatedName": "UserGetStreamResponse",
          "groupName": null,
          "nameOverride": null,
          "schema": "StreamUser",
          "type": "reference",
        },
        "type": "streamingJson",
      },
      "sdkName": {
        "groupName": [
          "user",
        ],
        "methodName": "get_stream",
      },
      "server": [],
      "summary": null,
      "tags": [],
    },
    {
      "audiences": [],
      "authed": false,
      "availability": null,
      "description": null,
      "errors": {},
      "examples": [
        {
          "codeSamples": [],
          "description": null,
          "headers": [],
          "name": null,
          "pathParameters": [
            {
              "name": "userId",
              "value": {
                "type": "primitive",
                "value": {
                  "type": "string",
                  "value": "userId",
                },
              },
            },
          ],
          "queryParameters": [],
          "request": {
            "properties": {
              "stream": {
                "type": "literal",
                "value": {
                  "type": "boolean",
                  "value": false,
                },
              },
            },
            "type": "object",
          },
          "response": {
            "properties": {
              "id": {
                "type": "primitive",
                "value": {
                  "type": "string",
                  "value": "id",
                },
              },
              "streamed": {
                "type": "literal",
                "value": {
                  "type": "boolean",
                  "value": false,
                },
              },
            },
            "type": "object",
          },
          "type": "full",
        },
      ],
      "generatedRequestName": "UserGetRequest",
      "headers": [],
      "idempotent": null,
      "internal": null,
      "method": "GET",
      "operationId": "get",
      "pagination": null,
      "path": "/user/{userId}",
      "pathParameters": [
        {
          "availability": null,
          "description": null,
          "name": "userId",
          "schema": {
            "availability": null,
            "description": null,
            "generatedName": "UserGetRequestUserId",
            "groupName": null,
            "nameOverride": null,
            "schema": {
              "default": null,
              "format": null,
              "maxLength": null,
              "minLength": null,
              "pattern": null,
              "type": "string",
            },
            "type": "primitive",
          },
          "variableReference": null,
        },
      ],
      "queryParameters": [],
      "request": {
        "additionalProperties": false,
        "contentType": null,
        "description": null,
        "schema": {
          "additionalProperties": false,
          "allOf": [],
          "allOfPropertyConflicts": [],
          "availability": null,
          "description": null,
          "generatedName": "UserGetRequest",
          "groupName": null,
          "nameOverride": null,
          "properties": [
            {
              "audiences": [
                "public",
              ],
              "availability": null,
              "conflict": {},
              "generatedName": "userGetRequestStream",
              "key": "stream",
              "nameOverride": null,
              "schema": {
                "availability": null,
                "description": null,
                "generatedName": "UserGetRequestStream",
                "groupName": null,
                "nameOverride": null,
                "type": "literal",
                "value": {
                  "type": "boolean",
                  "value": false,
                },
              },
            },
          ],
          "type": "object",
        },
        "type": "json",
      },
      "requestNameOverride": null,
      "response": {
        "description": "",
        "responseProperty": null,
        "schema": {
          "availability": null,
          "description": null,
          "generatedName": "UserGetResponse",
          "groupName": null,
          "nameOverride": null,
          "schema": "User",
          "type": "reference",
        },
        "type": "json",
      },
      "sdkName": {
        "groupName": [
          "user",
        ],
        "methodName": "get",
      },
      "server": [],
      "summary": null,
      "tags": [],
    },
  ],
  "globalHeaders": [],
  "groups": {},
  "hasEndpointsMarkedInternal": false,
  "idempotencyHeaders": [],
  "nonRequestReferencedSchemas": [
    "StreamUser",
    "User",
  ],
  "schemas": {
    "StreamUser": {
      "additionalProperties": false,
      "allOf": [],
      "allOfPropertyConflicts": [],
      "availability": null,
      "description": null,
      "generatedName": "StreamUser",
      "groupName": null,
      "nameOverride": null,
      "properties": [
        {
          "audiences": [],
          "availability": null,
          "conflict": {},
          "generatedName": "streamUserId",
          "key": "id",
          "nameOverride": null,
          "schema": {
            "availability": null,
            "description": null,
            "generatedName": "streamUserId",
            "groupName": null,
            "nameOverride": null,
            "type": "optional",
            "value": {
              "availability": null,
              "description": null,
              "generatedName": "StreamUserId",
              "groupName": null,
              "nameOverride": null,
              "schema": {
                "default": null,
                "format": "uuid",
                "maxLength": null,
                "minLength": null,
                "pattern": null,
                "type": "string",
              },
              "type": "primitive",
            },
          },
        },
        {
          "audiences": [],
          "availability": null,
          "conflict": {},
          "generatedName": "streamUserStreamed",
          "key": "streamed",
          "nameOverride": null,
          "schema": {
            "availability": null,
            "description": null,
            "generatedName": "streamUserStreamed",
            "groupName": null,
            "nameOverride": null,
            "type": "optional",
            "value": {
              "availability": null,
              "description": null,
              "generatedName": "StreamUserStreamed",
              "groupName": null,
              "nameOverride": null,
              "type": "literal",
              "value": {
                "type": "boolean",
                "value": true,
              },
            },
          },
        },
      ],
      "type": "object",
    },
    "User": {
      "additionalProperties": false,
      "allOf": [],
      "allOfPropertyConflicts": [],
      "availability": null,
      "description": null,
      "generatedName": "User",
      "groupName": null,
      "nameOverride": null,
      "properties": [
        {
          "audiences": [],
          "availability": null,
          "conflict": {},
          "generatedName": "userId",
          "key": "id",
          "nameOverride": null,
          "schema": {
            "availability": null,
            "description": null,
            "generatedName": "userId",
            "groupName": null,
            "nameOverride": null,
            "type": "optional",
            "value": {
              "availability": null,
              "description": null,
              "generatedName": "UserId",
              "groupName": null,
              "nameOverride": null,
              "schema": {
                "default": null,
                "format": "uuid",
                "maxLength": null,
                "minLength": null,
                "pattern": null,
                "type": "string",
              },
              "type": "primitive",
            },
          },
        },
        {
          "audiences": [],
          "availability": null,
          "conflict": {},
          "generatedName": "userStreamed",
          "key": "streamed",
          "nameOverride": null,
          "schema": {
            "availability": null,
            "description": null,
            "generatedName": "userStreamed",
            "groupName": null,
            "nameOverride": null,
            "type": "optional",
            "value": {
              "availability": null,
              "description": null,
              "generatedName": "UserStreamed",
              "groupName": null,
              "nameOverride": null,
              "type": "literal",
              "value": {
                "type": "boolean",
                "value": false,
              },
            },
          },
        },
      ],
      "type": "object",
    },
  },
  "securitySchemes": {
    "Bearer": {
      "headerEnvVar": "MY_API_TOKEN",
      "headerName": "Authorization",
      "headerVariableName": "token",
      "prefix": "Token",
      "type": "header",
    },
  },
  "servers": [],
  "tags": {
    "orderedTagIds": null,
    "tagsById": {},
  },
  "title": "Test extension \`x-fern-audiences\` alongside \`x-fern-streaming\`.",
  "variables": {},
  "webhooks": [],
}
`;
